lib/middlewares/object_scope_authorization.ex

```diff
@@ -64,97 +64,84 @@ defmodule Rajska.ObjectScopeAuthorization do
+    case is_authorized?(scope, result.root_value, context, type) do
+      true -> %{result | fields: field_result(fields, context)}
+      false -> Map.put(result, :errors, [error(emitter)])
+    end
   end
 
-  # When is an user defined object, lookup the scope meta tag.
-  defp authorize_object(object, fields, resolution, nested_keys) do
-    object
-    |> Type.meta(:scope)
-    |> is_authorized?(resolution, object, nested_keys)
-    |> put_result(fields, resolution, object, nested_keys)
+  # List
+  defp result(%{values: values} = result, context) do
+    %{result | values: list_result(values, context)}
   end
 
-  defp is_authorized?(nil, _, object, _nested_keys), do: raise "No meta scope defined for object #{inspect object.identifier}"
+  # Leafs
+  defp result(result, _context), do: result
 
-  defp is_authorized?(false, _resolution, _object, _nested_keys), do: true
+  # When is a list, inspect object that composes the list.
+  defp get_object_type(%Type.List{of_type: object_type}), do: object_type
+  defp get_object_type(object_type), do: object_type
 
-  defp is_authorized?({scoped_struct, field}, resolution, _object, nested_keys) do
-    field_keys = nested_keys ++ [field]
-    apply_authorization!(resolution, scoped_struct, Map.get(resolution, :value), field_keys)
-  end
+  defp field_result(fields, context, new_fields \\ [])
 
-  defp is_authorized?(scoped_struct, resolution, _object, nested_keys) do
-    apply_authorization!(resolution, scoped_struct, Map.get(resolution, :value), nested_keys ++ [:id])
-  end
+  defp field_result([], _context, new_fields), do: new_fields
 
-  defp apply_authorization!(resolution, scoped_struct, values, keys) when is_list(values) do
-    Enum.all?(values, fn value ->
-      apply_authorization!(resolution, scoped_struct, value, keys)
-    end)
+  defp field_result([field | fields], context, new_fields) do
+    new_fields = [result(field, context) | new_fields]
+    field_result(fields, context, new_fields)
   end
 
-  defp apply_authorization!(resolution, scoped_struct, nil, _keys) do
-    Rajska.apply_auth_mod(resolution, :has_resolution_access?, [resolution, scoped_struct, nil])
-  end
+  defp list_result(values, context, new_values \\ [])
 
-  defp apply_authorization!(resolution, scoped_struct, value, [first_key | remaining_keys]) when length(remaining_keys) > 0 do
-    nested_value = Map.get(value, first_key)
-    apply_authorization!(resolution, scoped_struct, nested_value, remaining_keys)
-  end
+  defp list_result([], _context, new_values), do: new_values
 
-  defp apply_authorization!(resolution, scoped_struct, value, [first_key]) do
-    scoped_field_value = Map.get(value, first_key)
-    Rajska.apply_auth_mod(resolution, :has_resolution_access?, [resolution, scoped_struct, scoped_field_value])
+  defp list_result([value | values], context, new_values) do
+    new_values = [result(value, context) | new_values]
+    list_result(values, context, new_values)
```

Comment:
What is the difference between `field_result/3` and `list_result/3`?

I believe they can be used interchangeably